### ISSUE TITLE
[BUG] Release notes for breaking change to `accept` in v12.0.0 are ambiguous

### ISSUE THREAD

**Issue by johnhunter** [react-dropzone/react-dropzone#1138]
**Describe the bug**

I don't think its clear what the breaking change to `accept` refers to in
https://github.com/react-dropzone/react-dropzone/releases/tag/v12.0.0

Does it, for example, involve a breaking change to the `accept` prop provided to `useDropzone` [documented here](https://github.com/react-dropzone/react-dropzone/blob/master/examples/accept/README.md). That uses `react-dropzone/attr-accept` which does not seem to have changed so I am confused by the release note.

**Expected**

Clear actions to take when a breaking change is reported in the package.

**Comment by rolandjitsu** [comment]
@johnhunter thanks for raising this point.

> I don't think its clear what the breaking change to accept refers to in

To me it seemed pretty obvious we're referring to the `accept` property you can set on the dropzone (either via the `useDropzone` hook or the class component `<Dropzone>`). There's no other `accept` in the context of react-dropzone.

I also tried to describe in [caveats#file-dialog-cancel-callback](https://github.com/react-dropzone/react-dropzone#file-dialog-cancel-callback), but I missed adding this in the [accept example](https://github.com/react-dropzone/react-dropzone/blob/master/examples/accept/README.md).

So I suppose we can update the docs to be more specific about it. And also mention that this limitation only applies when using click to show file picker in conjunction with `useFsAccessApi` (which is true by default).

> Clear actions to take when a breaking change is reported in the package.

There's nothing that the user can do besides disabling the FS access API usage via `useFsAccessApi` prop provided through `useDropzone({useFsAccessApi: false})`. The downside is that you get a less stable `onFileDialogCancel` cb.

On a side note, we've starting working on porting the docs to a different platform which hopefully will improve the overall structure as it's a bit all over the place at the moment.

**Comment by johnhunter** [comment]
Thanks for clarifying that. So to resolve the breaking change I would change as follows?:
```
useDropzone({
    accept: 'image/jpeg, image/png',
 });
```
It should change to:
```
useDropzone({
    accept: { 'image/*': ['.jpeg', '.jpg'] },
 });
```

**Comment by rpellerin** [comment]
Why does the code keep expecting a string or an array of strings? See https://github.com/react-dropzone/react-dropzone/blob/master/src/index.js#L99
![image](https://user-images.githubusercontent.com/5323445/152856194-6b8a7c8a-da5b-4733-9e45-a4cdbef11e35.png)

**Comment by rolandjitsu** [comment]
> Thanks for clarifying that. So to resolve the breaking change I would change as follows?:
> 
> ```
> useDropzone({
>     accept: 'image/jpeg, image/png',
>  });
> ```
> 
> It should change to:
> 
> ```
> useDropzone({
>     accept: { 'image/*': ['.jpeg', '.jpg'] },
>  });
> ```

Ok, no. Let me try to clarify. When you use `accept=".jpg"`, the file picker when triggered by click, it will not filter because the FS access API expects what you just posted above.

That's because we do not keep a map of file ext => MIME type. But if you pass mime types `accept="image/png"`, the file picker will work because we internally turn that string into the object that the FS access API needs:

https://github.com/react-dropzone/react-dropzone/blob/df8a9b4231b98fbc8b964c3cf0ca5f8682a50613/src/utils/index.js#L183

So, in summary:
1. the type for `accept` has not changed; it still takes a string or list of strings
2. in browsers that **do not support** the FS access API, the `accept` attr works exactly the same as in previous versions of react-dropozone, for both file picker and dag 'n' drop
3. in browser that **support** the FS access API, when passing file extensions to `accept` (e.g `".jpg,.png"`), the file picker will not apply those filters, but drag 'n' drop will
4. the use of FS access API can be turned off via `useDropzone({useFsAccessApi: false})`
